src/test/java/spoon/support/TypeAdaptorTest.java

```diff
@@ -440,4 +443,53 @@ public <T extends CharSequence> void overloaded(T t) {}
 
 		public <T extends String> void overriden(T t) {}
 	}
+
+	@GitHubIssue(issueNumber = 5226, fixed = true)
+	void testAdaptingTypeFromEnclosingClass() {
+		Launcher launcher = new Launcher();
+		launcher.getEnvironment().setComplianceLevel(11);
+		launcher.addInputResource("src/test/java/spoon/support/TypeAdaptorTest.java");
+		CtType<?> type = launcher.getFactory()
+				.Type()
+				.get(UseGenericFromEnclosingType.class);
+		@SuppressWarnings("rawtypes")
+		List<CtMethod> methods = type.getElements(new TypeFilter<>(CtMethod.class))
+				.stream()
+				.filter(it -> it.getSimpleName().equals("someMethod"))
+				.collect(Collectors.toList());
+		CtMethod<?> test1Method = methods.stream()
+				.filter(it -> !it.getDeclaringType().getSimpleName().startsWith("Extends"))
+				.findAny()
+				.orElseThrow();
+		CtMethod<?> test2Method = methods.stream()
+				.filter(it -> it.getDeclaringType().getSimpleName().startsWith("Extends"))
+				.findAny()
+				.orElseThrow();
+
+		assertTrue(test2Method.isOverriding(test1Method));
+		assertFalse(test1Method.isOverriding(test2Method));
+	}
+
+	public static class UseGenericFromEnclosingType {
```

Comment:
Do we want the classes here or in the resource folder?

Reply:
The other tests are also in this file, so I kept those close-by as well.